history/history-network.md

```diff
@@ -168,23 +165,40 @@ BlockHeaderWithProof = Container(
 )
 ```
 
+> **_Note:_** The `BlockHeaderProof` allows to provide headers without a proof (`None`).
+For pre-merge headers, clients SHOULD NOT accept headers without a proof
+as there is the `HistoricalHashesAccumulatorProof` solution available.
+For post-merge headers, there is currently no proof solution and clients MAY
+accept headers without a proof.
+
+##### Block Header by Hash
+
 ```python
 # Content and content key
 
 block_header_key = Container(block_hash: Bytes32)
 selector         = 0x00
 
-block_header_with_proof = BlockHeaderWithProof(header: rlp.encode(header)), proof: proof)
+block_header_with_proof = BlockHeaderWithProof(header: rlp.encode(header), proof: proof)
 
 content          = SSZ.serialize(block_header_with_proof)
 content_key      = selector + SSZ.serialize(block_header_key)
 ```
 
-> **_Note:_** The `BlockHeaderProof` allows to provide headers without a proof (`None`).
-For pre-merge headers, clients SHOULD NOT accept headers without a proof
-as there is the `HistoricalHashesAccumulatorProof` solution available.
-For post-merge headers, there is currently no proof solution and clients MAY
-accept headers without a proof.
+##### Block Header by Number
+
+
+```python
+# Content and content key
+
+block_number_key = Container(block_number: uint64)
+selector         = 0x04
```

Comment:
What's the reasoning not to reuse the 0x03 value of Epoch Records?

Avoiding some clashing of older versions that still decode this differently and having it more clear when something sends the old 0x03 data?

Reply:
> Avoiding some clashing of older versions that still decode this differently and it being more clear when something sends the old 0x03 data?

Initially

> What's the reasoning not to reuse the 0x03 value of Epoch Records?

I am fine reusing it, I wasn't sure if that was ok, I will update it now

Reply:
I'm not sure myself / have no strong preference.

With no reuse the transition period can be somewhat more clean in terms of error logs, and there are enough numbers available.
But I don't think it's really an issue to re-use either, and the end result is more clean :)

Reply:
I'm inclined towards re-using rather than leaving a gap.  Not a strong feeling.